### DISCUSSION TITLE
Mensa Page

### DISCUSSION THREAD

**Discussion by ghost** [astarub/campus_app#52] [Ideas]
1. I think it would be good to include a little dot that indicates if one of the places is closed, about to close or closed according to the opening times. This would mean a green dot could indicate the mensa is open a yellow dot (optional) could indicate that the mensa is about to close within the next 20 minutes and a red dot could indicate that the mensa is closed right now
2. Some Places are missing. For example Little Q could be listed with its location. Curry Q is also missing and maybe some others like the Sportler-Lounge. Imo Edwards, Beckmannshof, Cafeterias and similar should not yet be included.
3. The Opening Times or Locations could be listed as a section when extending the given place. This would make it possible to know the opening times of those places in general.
4. The filter is missing avoid vegetables
5. It should be explained what the App defines as vegetarian/vegan. This is neither intuitive nor obvious. This could be done in the Guide for example
6. Mensa can be called just Mensa or RUB Mensa to keep it cleaner.
7. The Places should be listed either Alphabetical->Opening Status or Opening Status -> Alphabetical currently it makes no Sense to list KulturCafé [...] Henkelmann.
8. The Kukaf list is too long when expended. It should either only list changing menu items like the mensa does or have a seperate button to open the menu and only show the opening times directly
9. If little Q is included it should have a show on map button and open the default map with its location.

**Comment by C4s4r** [comment]
Thanks for your several ideas! We will go through them one-by-one and discuss them in the next meeting.

Short feedback beforehand:
**2:** We already thought about scraping the positioning of Little Q, but we have to research this further before we can decide on that.
**7:** Students access this page pretty often during the week or even day. In my opinion, changing the positioning dynamically could lead to confusion as the user whould have to search the place they're looking for frequently.

**Reply by ghost** [reply]
7) id argue that youre mostly interested in what food is available a) today b) at a place where you can still get it. Might be an idea for future settings or could be reconsidered after asking users about it

**Comment by domai-tb** [comment]
Their all nice ideas and we should definitly implement nearly all of them!

1-3.) We currently start a cooperation with the AKAFÖ to get a handy data source, because we scrapping the conntent currently. This works... Not more. So we currently working on a reliable solution that we can access like an API and is maintained by AKAFÖ iteself to add support of all the other locations and information. But it will definitly depend on AKAFÖ too. 

5.) Can you explain what exactly you mean with "It should be explained what the App defines as vegetarian/vegan. This is neither intuitive nor obvious."? As mentioned above we scrapping the websites of AKAFÖ to get this information and don't add any of these by ourself. 

7.) I agree that our current ordering could be confusing. We should sort it in alphabetically order, but I wouldn't sort it by Opening status too. If we implement an traffic-ligth-dot, like you mentioned, this would be obsolet and could lead to more confusing like @C4s4r  said. 

8.) Yes its definitly too long. We are currently in discussions to add food labels too and will change the list at the same time. For now it is a hardcoded version. 

Cheers, :Domai

**Reply by ghost** [reply]
5) 
There are different definitions for the terms vegiterian and vegan. The Akafö defines vegan as:
"Unser Essen ist vegan wenn keine Zutat von folgenden Bestandteilen enthalten ist:
Fisch oder Spuren von Fisch Geflügel Lamm Rind Schwein Wild Krebstiere Eier Milch/ Laktose Weichtieren Gelatine Honig"

Some are not named and some are interpreted differently. For example clams mussels Butter Mayonnaise Honey Yeast food dye certain artificial ingredients.

There is no universal definition of what is vegan/vegetarian and what isnt. Thats why it should always be explained like the Akafö does on their page https://www.akafoe.de/gastronomie/haeufige-fragen.

Also the definitions from other places like the kukaf or future restaurants on campus might deverge from akafös definition. 

For example the European Vegan Union includes in the definition: 
"[...] Knochen oder tierische Fette, weder als Zutat noch als Hilfsmittel.
Zutaten oder Zusatzstoffe deren tierischer Ursprung nicht erkennbar ist.
Tierische Stoffe zur Klärung von Fruchtsäften, Essig, Wein.
Gentechnisch verändernten Organismen.
Zudem wird verlangt, dass alle Zutaten und Zusatzstoffe auf ihre Eigenschaften hin überprüft werden, damit jegliche Berührung mit tierischen Bestandteilen im Herstellungsprozess ausgeschlossen werden kann."
https://de.wikipedia.org/wiki/Veganismus#Kennzeichnung_und_Siegel

which is much stronger than the akafö ones.

**Reply by C4s4r** [reply]
I think at this point we should just refer to the AKAFÖ pages instead of trying to define that ourselves for each restaurant.

**Reply by ghost** [reply]
The point is that we have in either case to stick to one definition even if we pick the akafös one we have to apply it to the kukaf those deciding by ourselves what dishes are vegetarian and what arent as the kukaf will probably use different definitions and the same for upcoming restaurants that might be added.

Its just important to be transparent about the definition and let the user know what it actually means when a dish is marked as vegetarian/vegan. Without context its not that helpful to show in the first place.

We could also add a disclaimer for the given restaurant and show their definition.
